nrf_wifi/README.rst

```diff
@@ -6,6 +6,163 @@ nRF70 low-level API documentation
+
+.. note::
+
+   The command and event flow above is a simplified version of the actual flow, fragmented commands and events are not covered in this flow.
+
+Data flow: Transmit
+===================
+**FMAC** implements a basic token bucket algorithm to implement flowcontrol to the nRF70 Series Wi-Fi chipset.
+
+Buckets
+-------
+Both per-AC buckets and global buckets are implemented as follows:
+
+- **Per-AC buckets**: The nRF70 Series Wi-Fi chipset supports 5 Access Categories (ACs) (4 as per IEEE 802.11e and 1 for management frames).
+  The nRF70 Series Wi-Fi driver divides the tokens :kconfig:option:`CONFIG_NRF700X_MAX_TX_TOKENS` into equal parts for each AC.
+- **Global bucket** (a.k.a **Spare bucket**): The nRF70 Series Wi-Fi driver implements a global bucket that is shared between all ACs.
+  The remaining tokens after dividing the tokens for per-AC buckets are added to the global bucket.
+
+E.g. if :kconfig:option:`CONFIG_NRF700X_MAX_TX_TOKENS` is set to 12, the driver divides the tokens as follows:
+
+- 2 tokens for each AC
+- 2 tokens for the global bucket
+
+Queues
+------
+The nRF70 Series Wi-Fi driver implements the following queues:
+
+- **TX queue**: This queue is used to store the frames that are ready to be transmitted.
+  This queues maps to frames that can be transmitted in a single TXOP (Transmission Opportunity), e.g. a single A-MPDU.
+  The depth of this queue is controlled by the :kconfig:option:`CONFIG_NRF700X_MAX_TX_AGGREGATION` Kconfig option and also limited by the maximum size of memory for a single token.
+- **TX pending queue**: This queue is used to store the frames that are waiting for the tokens to be available in the buckets.
+   The depth of this queue is controlled by the :kconfig:option:`CONFIG_NRF700X_MAX_TX_PENDING_QLEN` Kconfig option.
+
+
+Flow
+----
+The nRF70 Series Wi-Fi driver implements the following flow for transmitting a frame:
+
+1. **FMAC** receives a frame from the networking stack.
+2. **FMAC** fetches the peer, TID and AC for the frame, depending on these the transmit pending queue is selected and the frame is enqueued.
+3. **FMAC** checks if the frame can be transmitted immediately, by checking
+   - If the frame meets the aggregation criterion, i.e. the maximum aggregation count is reached or maximum aggregation size is reached or frame cannot be aggregated.
+   - If the frame meets the token availability criterion, i.e. the frame can be transmitted based on the following:
+     - Availability of the per-AC and global tokens.
+4. If the frame can be transmitted, **FMAC** dequeues the frame from the transmit pending queue and enqueues it to the transmit queue.
+5. **FMAC** processes the transmit queue:
+   - prepares necessry information for **UMAC** firmware in the form of `CMD_TX` command.
+   - `copies` (this can be a memory copy or a DMA depending on underlying **BAL** implementation) each frame to the appropriate memory location in the Packet RAM
```

Comment:
```suggestion
   * It prepares the necessary information for **UMAC** firmware in the form of a ``CMD_TX`` command.
   * It copies (this can be a memory copy or a DMA depending on underlying **BAL** implementation) each frame to the appropriate memory location in the Packet RAM.
```